docs/creating/yaml.md

```diff
@@ -125,9 +206,160 @@ scenarios:
     * The imported files may include further files, so arbitrary degrees of nested configurations are possible.
     * The `import` statement can either give an absolute path or a path relative to the importing file.
 
-## Overriding one file with another
+### Reusing definitions through templates
```

Comment:
not needed, they can be used anywhere now
The code in nodes, techs and data_tables that handled this was removed in this PR.

Reply:
Correct. Your first example would work well.

I saw templates as purely 'extended' yaml. As in, they would work with any YAML file, and the templates code would live inside `io.py`.

You are also correct in that using templates within the override would introduce issues. My code currently does not cover for it. This is the kind of edge-case that is avoided by just solving it _before_ overrides.

So, two choices:
- Do we keep the current behaviour, and add code for the edge-case you described?
- Do we move templates to the `io.py` function (meaning that `read_rich_yaml` always solves them)?

I prefer the second even if we lose some templating power because it is more robust overall, code-wise.

Reply:
Correct. I will add a function that tests for that then.
It is important because after this goes in, schemas (`pydantic`) and so on must check the data structure of the dictionary returned by `model_definition`, so this edge case will be missed.

Reply:
After going through this edge-case, it seems like a non-issue. The templates algorithm will just see the 'new' `templates` or `template` requested if you change it on the data, and solve accordingly.

This is more a question of what would feel 'natural' to users: updating what `templates` 'do' within the overrides, or just seeing them as 'copy-paste' within the overrides. Either case should be basically the same in terms of the amount of bugs.

Updating the templates within the overrides seems too convoluted for me, so I would not use it, but I think it's fine to keep the code as-is.